### ISSUE TITLE
Graph shape constraint is too restrictive: data collector + semantic

### ISSUE THREAD

**Issue by azaroth42** [archesproject/arches#3010]
Related to #1480, but that is about UX not the actual implemented constraint.

### User Story
As a data modeler, I want to create a graph shape where a semantic node has a concept and a cardinality n semantic child. The semantic child would then collect data in a separate card, as the values are associated together and can be multiple.  This appears to violate the graph shape restrictions.

### Use Case:

A Name branch in which the Name can be composed of parts, and the set of parts for each name is not fixed, but can come from an RDM collection.

![arches-cards-issue-name](https://user-images.githubusercontent.com/871868/35753066-1b174a10-0813-11e8-9b0d-83cec9b6b6e8.png)

The left hand graph is the pattern we want, the right hand is an example instance.

### Impact:

All of our graphs require this sort of pattern, as everything can have a Name, and hence we're blocked on Provenance and AATA.

**Comment by azaroth42** [comment]
Another case -- if there are multiple actors with roles during an event like the production of an object, the same pattern occurs.

The top production is the overall production of the object, which then consists of the activities undertaken by each different actor.  The type on the lower level productions then lets you determine what sort of role was taken (painter vs scribe vs sculptor vs printer...)

![arches-production-card-issue](https://user-images.githubusercontent.com/871868/35759285-8e77e356-082d-11e8-860b-4b060c36aa4e.png)

**Comment by azaroth42** [comment]
And ... multiple middle names would need to have cardinality n for the middle-name typed Name part, even if the different chunks of name are broken out.  John Ronald Reuel Tolkien, for example.

Also the same appies for names of works with arbitrary numbers of subtitles or extra-bits-after-the-first-subtitle.

[Do I need to keep adding more use cases or is the point well enough made by now? 😀 ]

**Comment by dwuthrich** [comment]
@azaroth42 If this pattern is essential, then we will need to relax Arches' constraints on branch shapes.  We've looked at this and determined that we could support significantly more relaxed constraints in the graph designer and support arbitrarily deep nesting (one of our key concerns) in Arches' Card Manager and Resource Editor UI.

Another possible solution to your pattern (at least for names) could be as follows:
![names](https://user-images.githubusercontent.com/815185/35834743-e97781ac-0a8b-11e8-9a70-5e7a4e99f9af.png)

Pros: this pattern can be implemented in Arches without any issues.  It also supports multiple middle names with out having to pre-determine the naming fields (e.g.: it would support John Ronald Reuel Tolkien without any trouble).  

Potential downsides include splitting the "Aggregated Name" into one card, name parts into a separate card.  The semantic representation is a bit different.  But if its acceptable, this pattern might represent a work around for the cardinality =1 for aggregated (or parent) values and cardinality = n for related values.

After thinking about this issue for a bit, the idea of relaxing Arches branch shape constraints makes sense and might represent the better long-term solution.  On the other hand, if the pattern I've shown is acceptable, then we're no longer blocked.

**Comment by azaroth42** [comment]
> we could support significantly more relaxed constraints in the graph designer and support arbitrarily deep nesting  

👍 

As to the pattern described ... I don't think that the name identifies the name, or is the name of the name. Enumerating the fields will suffice for now, as the shape will remain the same in the future just that the model will be more flexible.

**Comment by azaroth42** [comment]
A side effect of the enumerate the possible components pattern is that when the data is exported, the enumerated bits are all included, even when they have no data.  For example, if the Name has a value, but none of the NameParts have any user entered information, the NameParts are still present in the graph.  In the model where you add NameParts as needed and configure them, this would not happen.

**Comment by apeters** [comment]
@azaroth42 

> For example, if the Name has a value, but none of the NameParts have any user entered information, the NameParts are still present in the graph.

I'm unsure if the above would be true if when you say "the NameParts are still present in the graph." you mean graph instance.  I don't think we currently export empty data structures.

**Comment by azaroth42** [comment]
I do mean graph instance, yes. I might have introduced the bug myself when exploring #3033 ... when there's the correct fix for that, I'll reverify :)

**Comment by azaroth42** [comment]
I think this can be closed in favor of the myriad issues in the Graph Manager Update work :)

**Comment by azaroth42** [comment]
I believe that this is fixed, and propose that we can close the issue.

**Comment by apeters** [comment]
per @azaroth42 comment above
